overlord/snapstate/conflict.go

```diff
@@ -110,14 +110,13 @@ func snapSetupFromChange(chg *state.Change) (*SnapSetup, error) {
 	return nil, nil
 }
 
-// verifySnapdIsNotRefreshing verifies that no change is currently running
-// for a snapd refresh/revert. While snapd is updating we do not allow other
-// snap changes to be executing, to avoid any incompatibilities in the task system,
-// between snapd versions.
+// changeIsSnapdDowngrade returns true if the change provided is a snapd
+// setup change with a version lower than what is currently installed. If a change
+// is not SnapSetup related this returns false.
 func changeIsSnapdDowngrade(st *state.State, chg *state.Change) (bool, error) {
 	snapsup, err := snapSetupFromChange(chg)
 	if err != nil || snapsup == nil || snapsup.SnapName() != "snapd" {
-		return false, nil
+		return false, err
```

Comment:
nitpick: I think this would be more readable if separating the `err != nil` from the other two cases so it is more easy to spot when an error is returned.

Reply:
Yea makes sense, thanks!